File: Data Structures & Algorithms/merge-intervals/submission-3.java
class Solution {
    public int[][] merge(int[][] intervals) {
         Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        int i=0,n=intervals.length ;
        while(i<n-1){
            if(intervals[i][1]>=intervals[i+1][0]){
                intervals[i][1]=Math.max(intervals[i][1],intervals[i+1][1]);
                ArrayList<int[]> list = new ArrayList<>(Arrays.asList(intervals));
                list.remove(i+1); 
                intervals = list.toArray(new int[0][]); 
                n=intervals.length;
            }
            else i++;
        }
        return intervals;
    }
}

File: Data Structures & Algorithms/non-overlapping-intervals/submission-3.js
class Solution {
    /**
     * @param {number[][]} intervals
     * @return {number}
     */
    eraseOverlapIntervals(intervals) {
        intervals.sort((a, b) => a[0] - b[0]);
        let count=0;
        let n=intervals.length ;
        for(let i=1;i<n;i++){
            if(intervals[i][0]<intervals[i-1][1]){
                count++;
                intervals[i][1]=Math.min(intervals[i][1],intervals[i-1][1]);
            }
        }
        return count;
    }
}
